apps/spark-3.5/src/test/java/com/linkedin/openhouse/catalog/e2e/SparkMoRFunctionalTest.java

```diff
@@ -0,0 +1,265 @@
+        tableName);
+  }
+
+  protected void createAndInitTable(String schema) {
+    sql("CREATE TABLE openhouse.%s (%s) USING iceberg %s", tableName, schema, "");
+    initTable();
+  }
+
+  @Test
+  public void testDeleteFilesCanBeCreated() throws NoSuchTableException {
+    createAndInitTable("id int, data string");
+
+    List<SimpleRecord> records =
+        Arrays.asList(
+            new SimpleRecord(1, "a"),
+            new SimpleRecord(1, "b"),
+            new SimpleRecord(1, "c"),
+            new SimpleRecord(2, "d"),
+            new SimpleRecord(2, "e"));
+    ops.spark()
+        .createDataset(records, Encoders.bean(SimpleRecord.class))
+        .coalesce(1)
+        .writeTo(tableName)
+        .append();
+
+    sql("DELETE FROM %s WHERE id = 1 and data='a'", tableName);
+    sql("DELETE FROM %s WHERE id = 2 and data='d'", tableName);
+    sql("DELETE FROM %s WHERE id = 1 and data='c'", tableName);
+
+    IcebergTableStats stats = ops.collectTableStats(tableName);
+    assertThat(stats.getNumPositionDeleteFiles() + stats.getNumEqualityDeleteFiles()).isEqualTo(3L);
+
+    List<Object[]> expected = Arrays.asList(row(1, "b"), row(2, "e"));
+    List<Object[]> actual = sql("SELECT * FROM %s ORDER BY id ASC", tableName);
+    assertThat(actual).containsExactlyElementsOf(expected);
+  }
+
+  @Test
+  public void testCompactionCanRemovePositionDeleteFiles() throws NoSuchTableException {
+    createAndInitTable("id int, data string");
+
+    List<SimpleRecord> records =
+        Arrays.asList(
+            new SimpleRecord(1, "a"),
+            new SimpleRecord(1, "b"),
+            new SimpleRecord(1, "c"),
+            new SimpleRecord(2, "d"),
+            new SimpleRecord(2, "e"));
+    ops.spark()
+        .createDataset(records, Encoders.bean(SimpleRecord.class))
+        .coalesce(1)
+        .writeTo(tableName)
+        .append();
+
+    sql("DELETE FROM %s WHERE id = 1 and data='a'", tableName);
+    sql("DELETE FROM %s WHERE id = 2 and data='d'", tableName);
+    sql("DELETE FROM %s WHERE id = 1 and data='c'", tableName);
+
+    Table table = ops.getTable(tableName);
+
+    // first run of compaction
+    RewriteDataFiles.Result result = rewriteFunc.apply(ops, table);
+    Assertions.assertEquals(1, result.addedDataFilesCount());
+    Assertions.assertEquals(1, result.rewrittenDataFilesCount());
+
+    // this asserts the dangling delete problem, since compaction did NOT remove the final delete
+    IcebergTableStats stats = ops.collectTableStats(tableName);
+    assertThat(stats.getNumPositionDeleteFiles()).isEqualTo(3L);
+    assertThat(stats.getNumCurrentSnapshotPositionDeleteFiles()).isEqualTo(1L);
+
+    // notice that running compaction again with no new commits cleans up the dangling delete
+    rewriteFunc.apply(ops, table);
+    stats = ops.collectTableStats(tableName);
+    assertThat(stats.getNumPositionDeleteFiles()).isEqualTo(3L);
+    assertThat(stats.getNumCurrentSnapshotPositionDeleteFiles()).isEqualTo(0L);
```

Comment:
I thought `rewrite_position_delete_files` running after `rewrite_data_files` is required to have this. Did we discover something different after we discussed this initially?

Reply:
yes, i thought the same. I assumed the dangling delete logic would still apply: compaction installs a new snapshot with the same data sequence number as the delete and thus the delete cannot be pruned.

 I had previously thought that a new data sequence number (a new snapshot with data) would need to be attached to the table and then a full table rewrite. 


what is happening in this test appears to conflict with the developer notes here:
https://docs.google.com/document/d/11d-cIUR_89kRsMmWnEoxXGZCvp7L4TUmPJqUC60zB5M/edit?tab=t.0#bookmark=id.cyzox4b19g6s


Perhaps we may be doing something wrong if we don't fully understand the results, in this case I plan to follow up with a more detailed validatino of the data sequence number and snapshot validation -- but due to time will focus on providing the feature as-is first.